torch/quantization/observer.py

```diff
@@ -195,27 +165,26 @@ def _calculate_qparams(self, min_val, max_val):
             else:
                 qmin, qmax = 0, 255
 
-        max_val, min_val = float(max_val), float(min_val)
-        min_val = min(0.0, min_val)
-        max_val = max(0.0, max_val)
-        if max_val == min_val:
-            scale = 1.0
-            zero_point = 0
+        min_val = torch.min(min_val, torch.zeros_like(min_val))
+        max_val = torch.max(max_val, torch.zeros_like(max_val))
+
+        scale = torch.ones(min_val.size(), dtype=torch.float32)
+        zero_point = torch.zeros(min_val.size(), dtype=torch.int64)
+
+        if self.qscheme == torch.per_tensor_symmetric or self.qscheme == torch.per_channel_symmetric:
+            max_val = torch.max(-min_val, max_val)
+            scale = max_val / (float(qmax - qmin) / 2)
+            scale = torch.max(scale, torch.tensor(1.0).new_full(scale.size(), self.eps, dtype=torch.float32))
```

Comment:
Why is this comparison needed? We do not need scale to be greater than 1.0.

Reply:
This is to make sure scale is greater than self.eps. Since scale is a tensor we need to create a tensor with value self.eps and the same shape as scale to compare with scale using torch.max(). I used new_full() on a dummy tensor with value 1.0 here to create such tensor, but there might be other ways to do it.